contracts/ethereum/contracts/UniswapOracleUtilityClient.sol

```diff
@@ -0,0 +1,190 @@
+import "./uniswap/UniswapLibraries.sol";
+
+contract UniswapOracleOverrideUtilityClient is IFluidClient, IEmergencyMode {
+    event DustCollected(address destination, uint amount);
+
+    IERC20 immutable token_;
+
+    uint immutable deltaWeightNum_;
+    uint immutable deltaWeightDenom_;
+
+    address immutable dustCollector_;
+
+    address private oracle_;
+    address private operator_;
+    address private emergencyCouncil_;
+
+    bool private noEmergencyMode_;
+
+    IUniswapV3Pool immutable pool_;
+
+    address immutable token0_;
+    address immutable token1_;
+
+    uint32 immutable secondsAgo_;
+
+    constructor(
+        IERC20 _token,
+        uint _deltaWeightNum,
+        uint _deltaWeightDenom,
+        address _dustCollector,
+        address _oracle,
+        address _operator,
+        address _council,
+        IUniswapV3Pool _pool,
+        uint32 _secondsAgo
+    ) {
+        token_ = _token;
+        pool_ = _pool;
+        secondsAgo_ = _secondsAgo;
+        deltaWeightNum_ = _deltaWeightNum;
+        deltaWeightDenom_ = _deltaWeightDenom;
+        dustCollector_ = _dustCollector;
+
+        oracle_ = _oracle;
+        operator_ = _operator;
+        emergencyCouncil_ = _council;
+        noEmergencyMode_ = true;
+
+        token0_ = pool_.token0();
+        token1_ = pool_.token1();
+
+        require(
+            address(_token) == token0_ || address(_token) == token1_,
+            "wrong pool!"
+        );
+
+        // might revert if the pool doesn't have enough data!
+        OracleLibrary.consult(address(pool_), _secondsAgo);
+    }
+
+    function drain() external {
+        require(msg.sender == operator_, "only operator");
+
+        uint balance = token_.balanceOf(address(this));
+
+        token_.transfer(dustCollector_, balance);
+
+        emit DustCollected(dustCollector_, balance);
+    }
+
+    // implements IFluidClient
+
+    /// @inheritdoc IFluidClient
+    function batchReward(Winner[] memory _rewards, uint _firstBlock, uint _lastBlock) external {
+        require(noEmergencyMode_, "emergency mode!");
+        require(msg.sender == oracle_, "only oracle");
+
+        uint poolAmount = token_.balanceOf(address(this));
+
+        for (uint i = 0; i < _rewards.length; i++) {
+            Winner memory winner = _rewards[i];
+
+            require(poolAmount >= winner.amount, "empty reward pool");
+
+            poolAmount = poolAmount - winner.amount;
+
+            token_.transfer(winner.winner, winner.amount);
+            emit Reward(
+                winner.winner,
+                winner.amount,
+                _firstBlock,
+                _lastBlock
+            );
+        }
+    }
+
+    /// @inheritdoc IFluidClient
+    function getUtilityVars() external view returns (UtilityVars memory) {
+        require(noEmergencyMode_, "emergency mode!");
+
+        (int24 arithmetic, /* uint128 harmonic */) = OracleLibrary.consult(
+            address(pool_),
+            secondsAgo_
+        );
```

Comment:
Imo should replace consult with an internal function with an application context